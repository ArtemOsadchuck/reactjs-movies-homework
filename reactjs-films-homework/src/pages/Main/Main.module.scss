$background-color: rgb(37, 37, 37);
$h2-loading-color: rgb(121, 121, 121);

.mainWrapper {
  align-items: center;
  background: $background-color;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 1100px;
  min-height: calc(100vh - 5rem);
  position: relative;
  width: 90%;

  .cardsWrapper {
    align-items: stretch;
    display: flex;
    flex-wrap: wrap;
    height: max-content;
    justify-content: space-around;
    margin-top: 4rem;
    width: 100%;
  }

  h2 {
    color: $h2-loading-color;
  }
}
