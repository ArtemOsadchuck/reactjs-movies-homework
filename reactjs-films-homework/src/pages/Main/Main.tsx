import React, { useEffect } from 'react';
import MovieCard from '../../components/MovieCard';
import { IMovieCard } from '../../components/MovieCard/MovieCard';
import CategoriesTabs from './CategoriesTabs';
import styles from './Main.module.scss';
import Pagination from './Pagination';
import { useAppSelector, useAppDispatch } from '../../hooks/hooks';
import getMainData from '../../store/rootStore/mainStore/getMainData';

const Main: React.FC = () => {
  const dispatch = useAppDispatch();
  const appFetchMovie = useAppSelector((state) => state.mainReducer.mainState);
  const mainState = {
    lang: useAppSelector((state) => state.mainReducer.lang),
    page: useAppSelector((state) => state.mainReducer.page),
    category: useAppSelector((state) => state.mainReducer.category),
    query: useAppSelector((state) => state.mainReducer.query),
  };

  useEffect(() => {
    dispatch(getMainData(mainState));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return appFetchMovie.length ? (
    <div className={styles.mainWrapper}>
      <CategoriesTabs />
      <div className={styles.cardsWrapper}>
        {appFetchMovie.length ? (
          appFetchMovie.map((e: IMovieCard) => {
            return <MovieCard props={e} key={Date.now() * Math.random()} />;
          })
        ) : (
          <h2>Loading...</h2>
        )}
      </div>
      <Pagination />
    </div>
  ) : (
    <div className={styles.mainWrapper}>
      return <h2>NO RESULTS FOUND</h2>;
    </div>
  );
};

export default Main;
