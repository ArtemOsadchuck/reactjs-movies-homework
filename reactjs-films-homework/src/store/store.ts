import rootStore from './rootStore/rootStore';

export type RootState = ReturnType<typeof rootStore.getState>;
export type AppDispatch = typeof rootStore.dispatch;
export default rootStore;
