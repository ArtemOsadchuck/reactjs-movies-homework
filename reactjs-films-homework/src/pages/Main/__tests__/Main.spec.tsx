import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import Main from '../Main';
import { Provider } from 'react-redux';
import store from '../../../store/store';

describe('Main', () => {
  let Fragment: any;
  beforeEach(() => {
    const asFragment = render(
      <Provider store={store}>
        <Main />
      </Provider>
    );
    Fragment = asFragment;
  });
  afterEach(() => {
    cleanup();
  });

  it('Main snapshot', () => {
    expect(Fragment).toMatchSnapshot();
    screen.debug();
  });

  it('Main heading must include text Loading...', () => {
    expect(screen.getByRole('heading')).toHaveTextContent('Loading...');
  });
  it('Main Pagination must include class active', () => {
    expect(screen.getByText(/1/i)).toHaveClass('active');
  });
  it('Main  must include inner text', () => {
    expect(screen.getAllByText(/Popular|Top|rated|Upcoming/i)).toBeDefined();
  });
});
