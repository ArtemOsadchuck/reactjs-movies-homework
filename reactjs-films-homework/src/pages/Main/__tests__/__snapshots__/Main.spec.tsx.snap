// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Main snapshot 1`] = `
<DocumentFragment>
  <div
    class="mainWrapper"
  >
    return 
    <h2>
      NO RESULTS FOUND
    </h2>
    ;
  </div>
</DocumentFragment>
`;
